tests/phpunit/tests/theme/wpGetGlobalStylesheet.php

```diff
@@ -270,4 +270,19 @@ public function test_should_enqueue_stored_styles() {
 			'Registered styles with handle of "wp-style-engine-my-styles" do not match expected value from the Style Engine store.'
 		);
 	}
+	/**
+	 * Tests that switching themes recalculates the stylesheet.
+	 *
+	 * @ticket 56970
+	 */
+	public function test_switching_themes_should_recalculate_stylesheet() {
+		$stylesheet_for_default_theme = wp_get_global_stylesheet();
+		switch_theme( 'block-theme' );
+		$stylesheet_for_block_theme = wp_get_global_stylesheet();
+
+		$this->assertStringNotContainsString( '--wp--preset--font-size--custom: 100px;', $stylesheet_for_default_theme, 'custom font size (100px) not present for default theme' );
+		$this->assertStringContainsString( '--wp--preset--font-size--custom: 100px;', $stylesheet_for_block_theme, 'custom font size (100px) is present for block theme' );
+
+		switch_theme( WP_DEFAULT_THEME );
```

Comment:
This should happen before the assertions in case either of them fails

Reply:
Ah, good point. I was misled by the other places in the code that does the same. Resolved at https://github.com/WordPress/wordpress-develop/pull/3712/commits/62518d145322c1d1e545398f496119f1612250fb